test/jdk/com/sun/security/auth/module/AllPlatforms.java

```diff
@@ -21,44 +21,39 @@
  * questions.
  */
 
-/*
+ /*
  * @test
- * @bug 8039951
+ * @bug 8039951 8281717
  * @summary com.sun.security.auth.module missing classes on some platforms
  * @run main/othervm AllPlatforms
  */
-
 import javax.security.auth.login.Configuration;
 import javax.security.auth.login.LoginContext;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 
 public class AllPlatforms {
+
+    private static final String OS = System.getProperty("os.name").toLowerCase();
+    private static final String UNIX_MODULE = "UnixLoginModule";
+    private static final String NT_MODULE = "NTLoginModule";
+
     public static void main(String[] args) throws Exception {
         login("cross-platform",
-                "UnixLoginModule", "optional",
-                "NTLoginModule", "optional",
-                "SolarisLoginModule", "optional");
-        try {
-            login("windows", "NTLoginModule", "required");
-            login("unix", "UnixLoginModule", "required");
-            login("solaris", "SolarisLoginModule", "required");
-        } catch (Exception e) {
-            e.printStackTrace(System.out);
-            if (e.toString().contains("UnsatisfiedLinkError")) {
-                throw new Exception("This is ugly");
-            }
-        }
+                UNIX_MODULE, "optional",
+                NT_MODULE, "optional");
+        login(OS.replaceAll("[^a-zA-Z0-9]", ""), getPlatformLoginModule(), "required");
```

Comment:
I think test should still be run on all platforms even though login would fail. On other platforms than the test being run on, test shouldn't fail with "UnsatisfiedLinkError" as per JDK-8039951.

Reply:
Now it will run in all platform other than the running one and the expected failure is LoginException. So any other types of Exception will be treated as failure including UnsatisfiedLinkError.

Reply:
thanks for addressing it!